src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -12227,7 +12230,42 @@ bool Assembler::reachable(AddressLiteral adr) {
   } else {
     disp += fudge;
   }
-  return is_simm32(disp);
+  return Assembler::is_simm32(disp);
+}
+
+bool Assembler::reachable(AddressLiteral adr) {
+  bool is_reachable = is_reachable_from(pc(), adr.target(), adr.reloc());
+  if (!is_reachable) {
+    assert(!always_reachable(adr), "sanity");
+  }
```

Comment:
Maybe a bit cleaner like this:

```suggestion
  assert(is_reachable || !always_reachable(adr), "sanity");
```